OpenSim/Simulation/Test/testInverseKinematicsSolver.cpp

```diff
@@ -0,0 +1,457 @@
+
+            cout << "time: " << state.getTime() << " | " << markerNames[2] 
+                << "(weight = " << markerWeights[2] << ") error = " 
+                << leftMarkerWeightedErrors[2] << endl;
+
+            // increasing the marker weight (marker[2] = "mL") should cause
+            //  that marker error to decrease
+            SimTK_ASSERT_ALWAYS(
+                leftMarkerWeightedErrors[2] < previousErr,
+                "InverseKinematicsSolver track failed to lower 'left' "
+                "marker error when marker weight was increased.");
+
+            previousErr = leftMarkerWeightedErrors[2];
+        }
+    }
+}
+
+Model* constructPendulumWithMarkers()
+{
+    Model* pendulum = new Model();
+    pendulum->setName("pendulum");
+    Body* ball =
+        new Body("ball", 1.0, SimTK::Vec3(0), SimTK::Inertia::sphere(0.05));
+    pendulum->addBody(ball);
+
+    // PinJoint hinge is 1m above ground origin and 1m above the ball in the ball
+    // reference frame such that the ball center is at the origin with the hinge
+    // angle is zero
+    PinJoint* hinge = new PinJoint("hinge", pendulum->getGround(),
+        SimTK::Vec3(0, 1.0, 0), SimTK::Vec3(0),
+        *ball, SimTK::Vec3(0, 1.0, 0), SimTK::Vec3(0));
+    hinge->updCoordinate().setName("theta");
+    pendulum->addJoint(hinge);
+
+    // Add Markers
+    Marker*m0 = new Marker();
+    m0->setName("m0");
+    m0->setParentFrame(*ball);
+    m0->set_location(SimTK::Vec3(0));
+    pendulum->addMarker(m0);
+    
+    // Shifted Right 1cm
+    Marker*mR = new Marker();
+    mR->setName("mR");
+    mR->setParentFrame(*ball);
+    mR->set_location(SimTK::Vec3(0.01, 0, 0));
+    pendulum->addMarker(mR);
+    
+    // Shifted Left 2cm
+    Marker*mL = new Marker();
+    mL->setName("mL");
+    mL->setParentFrame(*ball);
+    mL->set_location(SimTK::Vec3(-0.02, 0, 0));
+    pendulum->addMarker(mL);
+
+    return pendulum;
+}
+
+MarkerData* generateMarkerDataFromModelAndStates(const Model& model,
+    const StatesTrajectory& states, double noiseRadius, bool fixed)
+{
+    // use a fixed seed so that we can reproduce and debug failures.
+    std::mt19937 gen(0);
+    std::normal_distribution<double> noise(0.0, 1);
+
+    unique_ptr<Model> m{ model.clone() };
+    m->finalizeFromProperties();
+    
+    auto* markerReporter = new TableReporterVec3();
+    
+    auto markers = m->getComponentList<Marker>();
+    for (const auto& marker : markers) {
+        markerReporter->updInput().
+            connect(marker.getOutput("location"), marker.getName());
+    }
+
+    m->addComponent(markerReporter);
+
+    SimTK::State s = m->initSystem();
+
+    for (const auto& state : states) {
+        // collect results into reporter
+        m->realizeReport(state);
+    }
+
+    auto maxval = noise.max();
+
+    // make a copy of the reported table
+    auto results = markerReporter->getTable();
+
+    SimTK::Vec3 offset = noiseRadius*SimTK::Vec3(double(noise(gen)),
```

Comment:
Oh I see what's happening: the same offset is used for all markers and at all time steps if the `fixed` flag is set to `true`. A comment here would be helpful. Also perhaps consider changing the name of the flag from "fixed" to something more descriptive, like "constantOffset".